### PR TITLE
Fixes #2723: Disable post-code-deploy cloud hook for acsf builds.

### LINKED ISSUE #2723: post-code-deploy should not run on ACSF
My system information:
* BLT version: 9.0.5

Currently, it seems like best practice for ACSF customers is to set up both Cloud Hooks and ACSF Factory Hooks (`recipes:acsf:init:hooks` and `recipes:cloud-hooks:init`).

The problem is that the post-code-deploy hook doesn't really make sense on ACSF, because we already have the db-update factory hook that should take effect whenever someone explicitly deploys new code via the ACSF UI.

It can cause problems because it runs whenever you stage environments, when you probably _don't_ want hooks to be running.

On the other hand, post-code-update still makes sense to keep because I don't think there's any other way to automatically run updates when you push new code without explicitly starting a deploy via the ACSF UI.

### PR DESCRIPTION
**Pull request by malikkotob**
Fixes #2723

### CODE REVIEW

File: scripts/cloud-hooks/hooks/common/post-code-deploy/post-code-deploy.sh [hunk 1]
```diff
@@ -24,6 +24,9 @@ repo_root="/var/www/html/$site.$target_env"
 export PATH=$repo_root/vendor/bin:$PATH
 cd $repo_root
 
-blt artifact:ac-hooks:post-code-deploy $site $target_env $source_branch $deployed_tag $repo_url $repo_type --environment=$target_env -v --yes --no-interaction
+acsf_file="/mnt/files/$site.$target_env/files-private/sites.json"
+if [ ! -f $acsf_file ]; then
```
Comment (grasmash) [inline scripts/cloud-hooks/hooks/common/post-code-deploy/post-code-deploy.sh]: Let's keep logic inside PHP and outside of Bash wherever possible.

File: src/Robo/Commands/Artifact/AcHooksCommand.php [hunk 2]
```diff
@@ -33,7 +33,12 @@ class AcHooksCommand extends BltTasks {
    * @command artifact:ac-hooks:post-code-deploy
    */
   public function postCodeDeploy($site, $target_env, $source_branch, $deployed_tag, $repo_url, $repo_type) {
-    $this->postCodeUpdate($site, $target_env, $source_branch, $deployed_tag, $repo_url, $repo_type);
+    if (!$this->getInspector()->isAcsfInited()) {
```
[the remark this answers is not in the document]
Comment (malikkotob) [reply src/Robo/Commands/Artifact/AcHooksCommand.php]: @grasmash or @danepowell can you elaborate on this?
> Most ACSF enabled code bases are deployed on both ACSF and ACE.

[the remark this answers is not in the document]
Comment (malikkotob) [reply src/Robo/Commands/Artifact/AcHooksCommand.php]: Okay, that makes sense. Sounds like a new `isAcsfEnvironment` method on the inspector would be useful, then.

**Comment by malikkotob** [comment]
@grasmash should this be closed in favor of https://github.com/acquia/blt/pull/2768?

**Comment by grasmash** [comment]
Yes I think so.

**Pull request closed without merging**